Mathlib/CategoryTheory/Monoidal/Conv.lean

```diff
@@ -0,0 +1,62 @@
+/-
+Copyright (c) 2024 Lean FRO LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kim Morrison
+-/
+import Mathlib.CategoryTheory.Monoidal.Comon_
+
+/-!
+# The convolution monoid.
+
+When `M : Comon_ C` and `N : Mon_ C`, the morphisms `M.X ⟶ N.X` form a monoid (in Type).
+-/
+
+universe v₁ u₁
+namespace CategoryTheory
+open MonoidalCategory
+variable {C : Type u₁} [Category.{v₁} C] [MonoidalCategory C]
+
+/--
+The morphisms in `C` between the underlying objects of a pair of bimonoids in `C` naturally has a
+(set-theoretic) monoid structure. -/
+def Conv (M : Comon_ C) (N : Mon_ C) : Type v₁ := M.X ⟶ N.X
+
+namespace Conv
+
+variable {M : Comon_ C} {N : Mon_ C}
+
+instance : One (Conv M N) where
+  one := M.counit ≫ N.one
+
+@[simp, nolint simpNF] -- This simpNF incorrectly claims this simp lemma can not be applied.
+theorem one_eq : (1 : Conv M N) = M.counit ≫ N.one := rfl
+
+instance : Mul (Conv M N) where
+  mul := fun f g => M.comul ≫ f ▷ M.X ≫ N.X ◁ g ≫ N.mul
+
+@[simp] theorem mul_eq (f g : Conv M N) : f * g = M.comul ≫ f ▷ M.X ≫ N.X ◁ g ≫ N.mul := rfl
```

Comment:
Would not it be slightly better not to declare this as a global simp? (And similarly for `one_eq`.) Otherwise, when we do computations in this monoid, everything would be `dsimp`lified into these formulas, which would make the use of algebraic relations (associativity, etc) difficult. I would think that the user should explicitly trigger these lemmas when needed.

Reply:
Thank you, I agree.